src/Library/demos/Launcher/main.js

```diff
@@ -20,59 +12,56 @@ Gio._promisify(
 Gio._promisify(Gtk.FileDialog.prototype, "open", "open_finish");
 Gio._promisify(Gtk.UriLauncher.prototype, "launch", "launch_finish");
 
-const file_launcher = new Gtk.FileLauncher();
-const uri_launcher = new Gtk.UriLauncher();
-const file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
-  "Library/demos/Launcher/workbench.txt",
-);
-file_launcher.set_file(file);
-file_launcher.always_ask = true;
+const launch_file = workbench.builder.get_object("launch_file");
+const file_name = workbench.builder.get_object("file_name");
+const file_location = workbench.builder.get_object("file_location");
+const change_file = workbench.builder.get_object("change_file");
+const uri_launch = workbench.builder.get_object("uri_launch");
+const uri_details = workbench.builder.get_object("uri_details");
 
 //File Launcher
+
+const file = Gio.File.new_for_uri(workbench.resolve("workbench.txt"));
+const file_launcher = new Gtk.FileLauncher({
+  always_ask: true,
+  file,
+});
+
 launch_file.connect("clicked", () => {
   file_launcher.launch(workbench.window, null).catch(logError);
 });
 
 file_launcher.connect("notify::file", () => {
-  const file_info = file_launcher.get_file();
-  const details = file_info.query_info(
-    "standard::*",
-    Gio.FileQueryInfoFlags.NOFOLLOW_SYMLINKS,
+  const details = file_launcher.file.query_info(
+    "standard::display-name",
+    Gio.FileQueryInfoFlags.NONE,
     null,
   );
-  file_name.label = details.get_name();
+  file_name.label = details.get_display_name();
 });
 
 file_location.connect("clicked", () => {
   file_launcher.open_containing_folder(workbench.window, null).catch(logError);
 });
 
 change_file.connect("clicked", () => {
-  let new_file = null;
-
-  const dialog_for_file = new Gtk.FileDialog({
-    title: _("Select File"),
-    modal: true,
-  });
-  dialog_for_file
+  new Gtk.FileDialog()
     .open(workbench.window, null)
-    .then((result) => {
-      new_file = result;
-    })
-    .catch((err) => {
-      console.debug(err.message);
+    .then((file) => {
+      file_launcher.file = file;
     })
-    .finally(() => {
-      file_launcher.set_file(new_file);
```

Comment:
please use property getters/setters, but also, we don't want to do this when there is an error